### PR TITLE
Search refactor

### PR DESCRIPTION
**Pull request by andrewconner**
Fixes a few issues we've been having with the header. SearchCtrl now stays around as you type your search query. The UX should be a bit faster — keeps are only replaced when new results are available.

@stkem @ahsu1230 @yingjieMiao

### CODE REVIEW

File: kifi-backend/modules/shoebox/angular/src/search/search.tpl.html [hunk 1]
```diff
@@ -25,16 +25,4 @@
     toggle-edit="toggleEdit"
     update-selected-count="updateSelectedCount(numSelected)"
   ></div>
-  <div ng-if="!librariesEnabled"
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/search/search.tpl.html:28]: Should we start taking out these as we encounter them in our bug fixes?

File: kifi-backend/modules/shoebox/angular/src/layout/header/HeaderCtrl.js [hunk 2]
```diff
@@ -52,12 +52,13 @@ angular.module('kifi')
     });
     $scope.$on('$destroy', deregisterLibraryChip);
 
-    $scope.$on('$routeChangeSuccess', function (event, current) {
+    $scope.$on('$routeUpdate', function (event, current) {
```
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/angular/src/layout/header/HeaderCtrl.js:55]: Oh, so this calls even for page refreshes?

**Comment by ahsu1230** [comment]
:+1: thanks!

File: kifi-backend/modules/shoebox/angular/src/search/search.tpl.html (hunk 1 above)
Comment (stkem) [inline kifi-backend/modules/shoebox/angular/src/search/search.tpl.html:28]: Yes, I think we should

**Comment by stkem** [comment]
lgtm.
merging now to reduce merge conflict

File: kifi-backend/modules/shoebox/angular/src/search/search.tpl.html (hunk 1 above)
Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/src/search/search.tpl.html:28]: Let's do it.

Comment (ahsu1230) [inline kifi-backend/modules/shoebox/angular/src/search/search.tpl.html:28]: :ok_hand:

**Pull request merged**